### PR TITLE
Use stricter name validation creating from source repository

### PR DESCRIPTION
**Pull request by spadgett**
* Remove `osc-resource-name-validator` directive as different resources
  have different naming rules.
* Add stricter validation to the name field when creating from source
  repository to avoid errors with some resource types.

https://bugzilla.redhat.com/show_bug.cgi?id=1231107

### CODE REVIEW

File: assets/app/scripts/directives/oscResourceNameValidator.js [hunk 1]
```diff
@@ -6,7 +6,8 @@ angular.module("openshiftConsole")
     //github.com/GoogleCloudPlatform/kubernetes/pkg/util/validation.go
     //limiting to valid service names as LCD
     var maxNameLength = 24;
-    var VALID_NAME_RE = /^[a-z]([-a-z0-9]*[a-z0-9])?/i;
+    var minNameLength = 2;
+    var VALID_NAME_RE = /^[a-z]([-a-z0-9]*[a-z0-9])?$/;
```
Comment (liggitt) [inline assets/app/scripts/directives/oscResourceNameValidator.js]: these requirements can't be generalized to all resource types... some allow upper-case, some allow single-character names

Comment (liggitt) [inline assets/app/scripts/directives/oscResourceNameValidator.js]: most allow more than 24 characters

Comment (spadgett) [inline assets/app/scripts/directives/oscResourceNameValidator.js]: Hm, right now it's only see on this page. I could remove the directive and use `pattern`, `minlength`, and `maxlength` on the input.

Comment (liggitt) [inline assets/app/scripts/directives/oscResourceNameValidator.js]: yeah, we can't do a general one at this point

File: assets/app/scripts/directives/oscResourceNameValidator.js [hunk 2]
```diff
@@ -1,35 +0,0 @@
...
-      require: "ngModel",
-      link: function(scope, elm, attrs, ctrl) {
-        ctrl.$validators.oscResourceNameValidator = function(modelValue, viewValue){
-          if(ctrl.$isEmpty(modelValue)){
-            return false;
-          }
-          if(viewValue === null){
-            return false;
-          }
-          if(ctrl.$isEmpty(viewValue.trim())){
```
Comment (liggitt) [inline assets/app/scripts/directives/oscResourceNameValidator.js]: lost trim behavior... do we care?

**Comment by liggitt** [comment]
LGTM, @smarterclayton merge?

**Comment by smarterclayton** [comment]
Approved [merge

**Comment by smarterclayton** [comment]
[merge]

**Comment by smarterclayton** [comment]
Really?

**Comment by liggitt** [comment]
helps me know what to babysit in my list of assigned PRs

File: assets/app/scripts/directives/oscResourceNameValidator.js (hunk 2 above)
Comment (spadgett) [inline assets/app/scripts/directives/oscResourceNameValidator.js]: Looks like it only trims to validate if the field is empty, in which case our regex would catch it.

**Pull request merged**